pkg/isoeditor/nmstate_handler_test.go

```diff
@@ -71,12 +72,188 @@ var _ = Context("with test files", func() {
 		})
 
 		It("ram disk created successfully", func() {
-			err := nmstateHandler.CreateNmstateRamDisk("", ramDiskPath, nmstatectlPathForCaching)
+			// Pre-create cached nmstatectl file (as would be done by ExtractAndCacheNmstatectl in normal flow)
+			cachedContent := []byte("test-nmstatectl-binary")
+			err := os.WriteFile(nmstatectlPathForCaching, cachedContent, 0755) //nolint:gosec
+			Expect(err).NotTo(HaveOccurred())
+
+			err = nmstateHandler.CreateNmstateRamDisk("", ramDiskPath, nmstatectlPathForCaching)
 			Expect(err).ToNot(HaveOccurred())
 
 			exists, err := fileExists(ramDiskPath)
 			Expect(err).ToNot(HaveOccurred())
 			Expect(exists).To(BeTrue())
 		})
+
+		It("uses cached nmstatectl when available", func() {
```

Comment:
This tests the caching so the previous test doesn't need the cached content added before the call.